python/dgl/heterograph.py

```diff
@@ -6065,6 +6065,88 @@ def local_scope(self):
         self._node_frames = old_nframes
         self._edge_frames = old_eframes
 
+    def new_formats(self, allowed_formats=None, create_formats=None):
+        r"""Get a cloned graph with the specified allowed sparse format(s)
+        and create specific sparse matrices allowed for the graph or query
+        for the usage status of sparse formats.
```

Comment:
Do we still want to overload a single API for both setting and logging?

Reply:
Instead of having a api, maybe it is better to have an attribute for formats?

Reply:
Yes, could be.